File: 204-count-primes/count-primes.java
class Solution {
    public int countPrimes(int n) {
        int count = 0;
        boolean b[] = new boolean[n + 1];
        for (int i = 2; i * i <= n; i++) { 
            if (!b[i]) { 
                for (int j = i * i; j <= n; j += i) { 
                    b[j] = true; 
                }
            }
        }
        for (int i = 2; i < n; i++) {
            if (!b[i])
                count++; 
        }        
        return count;
    }
}

File: 204-count-primes/count-primes.py
class Solution:
    def countPrimes(self, n: int) -> int:
        if n <= 2:
            return 0
        
        b = [False] * n  
        count = 0
        
        for i in range(2, int(n ** 0.5) + 1):
            if not b[i]:
                for j in range(i * i, n, i):
                    b[j] = True
        
        for i in range(2, n):
            if not b[i]:
                count += 1
        
        return count